rmw_opensplice_cpp/src/rmw_node.cpp

```diff
@@ -418,6 +418,29 @@ rmw_destroy_node(rmw_node_t * node)
   return result;
 }
 
+rmw_ret_t
+rmw_node_assert_liveliness(const rmw_node_t * node)
+{
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_INVALID_ARGUMENT);
+
+  auto node_info = static_cast<OpenSpliceStaticNodeInfo *>(node->data);
+  if (nullptr == node_info) {
+    RMW_SET_ERROR_MSG("node info handle is null");
+    return RMW_RET_ERROR;
+  }
+  if (nullptr == node_info->participant) {
+    RMW_SET_ERROR_MSG("node internal participant is invalid");
+    return RMW_RET_ERROR;
+  }
+
+  if (node_info->participant->assert_liveliness() != DDS::RETCODE_OK) {
```

Comment:
I believe there is already a function for checking a DDS error code that returns an RMW error code. I would prefer we run the output through that and return the translated RMW code instead of just returning `RMW_RET_ERROR`. It might not make much difference at the moment, but it will make it more flexible going forward.

Reply:
Can you point me to this function? I couldn't find it.

Reply:
@mm318 take a look at rmw_opensplice_cpp/src/event_converter.cpp

```
rmw_ret_t check_dds_ret_code(DDS::ReturnCode_t & dds_return_code)
```

Reply:
Done.